src/activity_actor.cpp

```diff
@@ -6656,7 +6466,7 @@ void prying_activity_actor::handle_prying( Character &who )
             return;
         }
 
-        const furn_str_id new_furn = furn_type->prying->result();
+        const furn_str_id new_furn = std::get<furn_str_id>( furn_type->prying->result() );
         if( !new_furn.is_valid() ) {
             if( !testing ) {
                 debugmsg( "prying furniture: %s invalid furniture", new_furn.str() );
```

Comment:
`handle_prying` is exceptional and I did not deduplicate it. It has some internal functionality including calls to lambdas for difficulty and failure results, which do not appear in any other `::finish` method. The general finish method could take this sort of thing as a function parameter, sacrificing readability of that method for deduplicating this one. I chose to only make that trade where it affected most of the methods, not just one.